src/v2/Apps/Show/ShowApp.tsx

```diff
@@ -26,13 +26,20 @@ import {
   SharedArtworkFilterContextProps,
 } from "v2/Components/ArtworkFilter/ArtworkFilterContext"
 import { RouterLink } from "v2/System/Router/RouterLink"
+import { ShowNavigationBannerFragmentContainer as ShowNavigationBanner } from "./Components/ShowNavigationBanner"
+import { useRouter } from "v2/System/Router/useRouter"
 
 interface ShowAppProps {
   show: ShowApp_show
 }
 
 export const ShowApp: React.FC<ShowAppProps> = ({ show }) => {
   const { contextPageOwnerSlug, contextPageOwnerType } = useAnalyticsContext()
+  const { match } = useRouter()
+  const { from_fair } = match.location.query
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const arrivedFromFair = useMemo(() => from_fair ?? false, [])
```

Comment:
**suggestion**: I think it is not necessary to use `useMemo` here

```javascript
const arrivedFromFair = from_fair ?? false
```

Reply:
I wondered about that too - I think the idea is that we don't need to recompute `arrivedFromFair` on every re-render because `from_fair` will not change the whole time we're on the page. So seems like a slight performance optimization?

Reply:
There is no point in optimization here, since either `true` or `false` is expected all the time

Reply:
When we use `useMemo` we compute `arrivedFromFair` only one time and then even if user click "Back to _Fair_" the banner are not disappeared

Reply:
Aah, I totally forgot about it. Thank you for the reminder!